### DISCUSSION TITLE
ippeveprinter get always: Unable to create server credentials

### DISCUSSION THREAD

**Discussion by sbamamoto** [OpenPrinting/cups#992] [Q&A] [answered]
Hi,

starting ippeveprinter like this:

    /usr/bin/ippeveprinter -d /var/lib/myjobs -k -c /var/lib/myprinter/printHook.sh -p 10001 -P /var/lib/myprinter/printer.ppd  PRN

When I try to use the Bonjour announcement of the printer to add the printer in MAC OS always get this error:

    Unable to encrypt connection: Unable to create server credentials.

When I add the printer manually the printer works as expected.

My question is what can I do to make enable ippeveprinter to create the needed server credentials or is there another workaround to prevent this when using Bonjour ?

**Comment by michaelrsweet** [comment]
What version of CUPS are you using?

**Reply by sbamamoto** [reply]
I use debian 12 and installed this package:

cups-ipp-utils 2.4.2-3+deb12u5

**Reply by michaelrsweet** [reply, marked as answer]
OK, so if you are running ippeveprinter as root type:

```
sudo mkdir -p /etc/cups/ssl
```

For a regular user account type:

```
mkdir -p ~/.cups/ssl
```

Newer versions of CUPS actually try to make the corresponding directories.

**Reply by sbamamoto** [reply]
Thanks a lot worked like a charm. Do you have hint for me where to find such information about ippeveprinter? I tried to find this information by my self but failed.

**Reply by michaelrsweet** [reply]
We don't have any documentation detailing where certificates are stored, mainly because the location depends on the OS and configuration. Future versions of CUPS may make this less variable, at which point then we'd document it. But in any case we now create the directories as needed so you won't run into this issue.
